### DISCUSSION TITLE
Component proposal: CUI Banner

### DISCUSSION THREAD

**Discussion by amyleadem** [uswds/uswds#5767] [Proposals]
# CUI Banner proposal

### Component name

Controlled Unclassified Information (CUI)  Banner

### Description

The CUI banner creates a consistent pattern for marking controlled unclassified information in electronic contexts.

### Design system need

CUI must be clearly marked when it is displayed on a website or printed from the screen. Creating a common CUI banner would create a clear and consistent pattern that ensures compliance with current policy.

From GSA’s CUI Guide: “CUI indicators must be present to alert users of the presence of CUI within the system. A warning must be shown on the login screen or via a screen after logging in, and optionally in headers that appear on each page displayed. Additionally, any printouts generated from an application must have the CUI banner on every page of the printout. If these are not printed by the application, the user must write the banner on every page that is printed or use a CUI cover sheet to protect the information that is printed.”

#### Related laws and policies

The CUI program was established by Executive Order 13556 [PDF], and implemented by 32 CFR part 2002.

### Component design

The CUI banner should clearly convey to users that they are viewing controlled unclassified information.

The banner must be present both when viewed on screen and when printed from the screen.

#### Resources

- [CUI Marking guide](https://www.archives.gov/files/cui/documents/20161206-cui-marking-handbook-v1-1-20190524.pdf) [NARA]
- [CUI Marking quick tips](https://www.gsa.gov/system/files/CUI%20Quick%20Tips.docx) [GSA]

#### Possible variants

There are two likely variants for this component:

- **CUI Basic:** CUI Basic is, as the name implies, the standard “flavor” of CUI. All of the rules of CUI apply to CUI Basic Categories and Subcategories, making the handling and marking of CUI Basic the simplest. [NARA’s CUI marking guide](https://www.archives.gov/files/cui/documents/20161206-cui-marking-handbook-v1-1-20190524.pdf).
- **CUI Specified:** CUI Specified is different, since the requirements for how users must treat each type of information vary with each Category or Subcategory. This is because some Authorities have VERY specific requirements for how to handle the type of information they pertain to – requirements that simply would not make sense for the rest of CUI. CUI Specified is NOT a “higher level” of CUI, it is simply different.
  And because the things that make it different are dictated in laws, Federal regulations, and government-wide policies, they are not things that can legally be ignored or overlooked. As such, a document containing multiple CUI Specified Categories and Subcategories must include ALL of them in the CUI Banner Marking. [NARA’s CUI marking guide](https://www.archives.gov/files/cui/documents/20161206-cui-marking-handbook-v1-1-20190524.pdf)

#### Demonstration

Here is a simple mockup of a CUI banner from NARA’s CUI marking guide:

### Implementation examples

Provide a list of links to examples of this kind of component on government (preferred) or non-government sites.

### Related USWDS components

This component could possibly be a cousin of the site alert component.

### Key audiences

This component is designed for agencies that want to display CUI on websites and applications.

### When to use this component

- Use this component when you need to mark controlled unclassified information.
- Follow the specific guidelines for the particular type of CUI that is present (CUI Basic/CUI Specified)
- All other rules and regulations for CUI must be followed.

### When to consider a different component

Do not use this component for content that does not contain CUI.

### Usability considerations

The banner should be prominent and clearly communicate its message.

### Accessibility considerations

The banner should be visible to all assistive devices. The accessibility guidance added to the site alert component could possibly apply here.

### Volunteers

If there are any volunteers who are willing to help design, develop, or test this component, tag them here with details about their roles.

### Endorsements

If there is significant support from a given agency or group, include their names here and tag any stakeholders.

### Contributors

Create a list of those who helped carry out the research and work to prepare this proposal. Include your name and/or GitHub username if you'd like credit as a contributor.

**Comment by mshep02** [comment]
I'm happy to help you do research for this. I have a list of sources already compiled.

**Reply by mejiaj** [reply]
CC @jaclinec

**Reply by jaclinec** [reply]
@mshep02 we'd love to collaborate and see the sources you've compiled. Feel free to link or list here if able. :)

**Comment by sknep** [comment]
A few years ago, I worked on [Astro](https://github.com/RocketCommunicationsInc/astro/tree/main/packages/web-components/src/components/rux-classification-marking), a design system for the Air Force Space and Missile Systems Center, where we developed these classification marking components: https://www.astrouxds.com/components/classification-markings/

**Reply by amalyah** [reply]
Exactly what I was going to post, @sknep! :) (Also, hello!)
